### DISCUSSION TITLE
One to Many with one active

### DISCUSSION THREAD

**Discussion by petr-zeman** [symfony/symfony#49724] [Q&A]
Hi.

I would like to ask what is the best approach for this in Symfony.

E.g. there is one entity **_User_** with the One to Many relation to another entity **_Address_**. 
Let's say the **_User_** can have multiple delivery addresses in the e-shop, while exactly one is active as preset, with the possibility to choose other stored addresses. Or "_the active address_" is seen as the billing address.

Without Symfony, I could use e.g. stored procedure or the view with two possibilities.
Having another column/property of the _**User**_, holding ID of "_the active address_" record, or having another (e.g. boolean) column/property of the **_Address_** saying if the record is the active one. Then the view or stored procedure can easily return a single record containing the **_User_** data including "_the active address_". It would depend on other requirements, e.g. if it's needed to get the list of all active addresses, without the need of joining a **_User_** table/entity. 

How this could be achieved in the Symphony the clean way according to best practices?

And "a less urgent" question, by the way. Does Symfony / Doctrine support some preferred way to work with stored procedures? E.g. when complete business logic, including validations, access rights, etc. is implemented at the DB level and multiple clients (e.g. desktop app, web app, mobile app, ...) are using stored procedures instead of reimplementing business logic in each of them?

**Comment by MatTheCat** [comment]
Hi,

not sure to understand everything you ask but using Doctrine ORM [you don’t work directly with foreign keys but with references to the object they identify](https://www.doctrine-project.org/projects/doctrine-orm/en/2.14/reference/association-mapping.html).

So you could have a reference to the active address in your `User` entity, whose access wouldn’t load the addresses collection but only the active one.

> Does Symfony / Doctrine support some preferred way to work with stored procedures?

Depending on your needs it seems you could use the ORM’s [`NativeQuery`](https://www.doctrine-project.org/projects/doctrine-orm/en/2.14/reference/native-sql.html) or work lower-level using [DBAL’s connection](https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/data-retrieval-and-manipulation.html#data-retrieval-and-manipulation).
